### PR TITLE
Upgraded project to RC2. Replaced version in global.json with RC2. Up…

### PR DESCRIPTION
**Pull request by kg73**
Updated global.json to RC2's sdk version. updated project.json to target moniker 'net451' instead of the deprecated 'dnx451'. Updated version number of package in project.json to '1.0.0-rc2'

### CODE REVIEW

File: src/IdentityServer3.Integration.AspNet/project.json [hunk 1]
```diff
@@ -1,19 +1,20 @@
...
-    "dnx451": {
+    "net451": {
       "dependencies": {
-        "IdentityServer3": "2.1.1",
-        "Microsoft.Owin": "3.0.1",
-        "Microsoft.AspNet.DataProtection": "1.0.0-rc1-final",
-        "Microsoft.AspNet.Owin": "1.0.0-rc1-final"
+      "IdentityServer3": "2.1.1",
+      "Microsoft.Owin": "3.0.1",
+      "Microsoft.AspNet.DataProtection": "1.0.0-rc1-final",
```
Comment (JakeGinnivan) [inline src/IdentityServer3.Integration.AspNet/project.json]: These dependencies should probably be updated for rc2 too

**Comment by leastprivilege** [comment]
Maybe someone here can answer my question - 

why would you use idsrv3 in an aspnetcore host? idsrv3 is optimized and tested for Katana and runs thousands of times in production in that configuration.

idrsrv4 when it's done is for aspnetcore. 

Help me understand.

**Comment by kg73** [comment]
I need WS-Federation, but it is not yet support in idsrv4 due to not having support in asp.net core. Based on [this issue](https://github.com/aspnet/Security/issues/605), it appears ws-federation will not be supported in the initial release of asp.net core. 

Long term, I will switch to use idsrv4 once all the pieces are available. Until then, this integration library allows me to use idsrv3 and ws-federation on asp.net core.

**Comment by JakeGinnivan** [comment]
Well, for me I wanted Identity 3 and aspnet core/mvc6/whatever the version is and it made sense to use the same version of MVC for the API we are starting. 

If we built on owin and the old stack it means the deployment/hosting of the IDS3 and our API would be different. Am on the fence to switch IDS3 back to owin rather than building on this stack but TBH apart from a few little issues it's worked great.

**Comment by leastprivilege** [comment]
Jake, 

so you are saying the integration works well? What are the gotchas then? I can release it if you like - but I want to add a disclaimer to it.

**Comment by JakeGinnivan** [comment]
Biggest one I hit was plumbing in a customer `IUserService`, I fixed with it this ugly bit of code:

```
Factory = new IdentityServerServiceFactory
{
    UserService = new Registration<IUserService>(r =>
    {
        var owin1 = r.Resolve<OwinEnvironmentService>();
        var httpContext1 = (HttpContext) owin1.Environment["Microsoft.AspNetCore.Http.HttpContext"];
        return httpContext1.RequestServices.GetRequiredService<CustomUserService>();
    })
}
```

This PR is not enough to release for RC2 because the actual integration has changed slightly. Once the libraries are upgraded to the RC2 libraries in addition to the changes here this is what I ended up with:

```
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using IdentityServer3.Core.Configuration;
using Microsoft.AspNetCore.Builder;
using Microsoft.AspNetCore.DataProtection;
using Microsoft.Extensions.DependencyInjection;
using DataProtectionProviderDelegate = System.Func<string[], System.Tuple<System.Func<byte[], byte[]>, System.Func<byte[], byte[]>>>;
using DataProtectionTuple = System.Tuple<System.Func<byte[], byte[]>, System.Func<byte[], byte[]>>;
using Owin;

namespace ApiInfrastructure
{
    // TODO Replace with https://github.com/IdentityServer/IdentityServer3.Integration.AspNetCore
    // When it supports RC2
    public static class IdentityServerApplicationBuilderExtensions
    {
        public static void UseIdentityServer(this IApplicationBuilder app, IdentityServerOptions options)
        {
            app.UseOwin(addToPipeline =>
            {
                addToPipeline(next =>
                {
                    var builder = new Microsoft.Owin.Builder.AppBuilder();
                    var provider = app.ApplicationServices.GetRequiredService<IDataProtectionProvider>();

                    builder.Properties["security.DataProtectionProvider"] = new DataProtectionProviderDelegate(purposes =>
                    {
                        var dataProtection = provider.CreateProtector(string.Join(",", purposes));
                        return new DataProtectionTuple(dataProtection.Protect, dataProtection.Unprotect);
                    });

                    builder.UseIdentityServer(options);

                    var appFunc = builder.Build(typeof(Func<IDictionary<string, object>, Task>)) as Func<IDictionary<string, object>, Task>;
                    return appFunc;
                });
            });
        }
    }
}
```

Also has some smaller issues around how to resolve the path of the test cert in development and such but most of those issues are more about adjusting to changes in bootstrapping a aspnet core project.

**Comment by leastprivilege** [comment]
could you send a new PR?

**Comment by JakeGinnivan** [comment]
Sure

**Comment by leastprivilege** [comment]
thanks

**Pull request closed without merging**